src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -761,17 +762,118 @@ impl CloudHypervisorInner {
     }
 
     pub(crate) fn set_guest_memory_block_size(&mut self, size: u32) {
-        self._guest_memory_block_size_mb = size;
+        self.guest_memory_block_size_mb = size;
     }
 
     pub(crate) fn guest_memory_block_size_mb(&self) -> u32 {
-        self._guest_memory_block_size_mb
+        self.guest_memory_block_size_mb
     }
 
-    pub(crate) fn resize_memory(&self, _new_mem_mb: u32) -> Result<(u32, MemoryConfig)> {
-        warn!(sl!(), "CH memory resize not implemented - see https://github.com/kata-containers/kata-containers/issues/8801");
+    pub(crate) async fn resize_memory(
+        &mut self,
+        mut new_mem_mb: u32,
+    ) -> Result<(u32, MemoryConfig)> {
+        // check for probe in memory_config
+        let caps = &self.capabilities;
+        if caps.is_mem_hotplug_probe_supported() {
+            warn!(sl!(), "probe memory is not supported for cloud hypervisor");
+            return Ok((0, MemoryConfig::default()));
+        }
+
+        // check for resizing to 0
+        if new_mem_mb == 0 {
+            error!(sl!(), "Cannot resize memory to 0");
+            return Ok((0, MemoryConfig::default()));
+        }
+
+        // get hotplug size
+        let max_hotplug_size_mb = self.hypervisor_config().memory_info.default_maxmemory;
+        // do we want to set new_mem_mb to hotplug_size or fail?
+        if new_mem_mb > max_hotplug_size_mb {
+            warn!(sl!(), "Memory size unchanged; requested memory size {} is greater than max memory size {}", new_mem_mb, max_hotplug_size_mb);
+            return Ok((0, MemoryConfig::default()));
+        }
+
+        // get mem size
+        let default_mem_size_mb = self.hypervisor_config().memory_info.default_memory;
+        let current_hotplug_size_mb = self.mem_hotplug_size_mb;
+        let current_mem_size_mb = default_mem_size_mb + current_hotplug_size_mb;
+
+        if new_mem_mb == current_mem_size_mb {
+            info!(sl!(), "VM already has requested memory size {}", new_mem_mb);
+            return Ok((current_mem_size_mb, MemoryConfig::default()));
+        }
+
+        if new_mem_mb < current_mem_size_mb {
+            warn!(sl!(), "Remove memory is not supported, nothing to do");
+            return Ok((current_mem_size_mb, MemoryConfig::default()));
+        }
+
+        let block_size_mb = self.guest_memory_block_size_mb;
+
+        let new_hotplug_size_mb = new_mem_mb - current_mem_size_mb;
+        let aligned_new_hotplug_size_mb =
+            checked_next_multiple_of(new_hotplug_size_mb.into(), block_size_mb.into());
+
+        if let Some(aligned_value_mb) = aligned_new_hotplug_size_mb {
+            let aligned_request = current_mem_size_mb + aligned_value_mb as u32;
+            if new_mem_mb != aligned_request {
+                info!(
+                    sl!(),
+                    "Aligning VM memory request {} to {}", new_mem_mb, aligned_request
```

Comment:
Provided we have to have this log message, I'd propose including `block_size_mg` as well so that complete information related to aligning is available to anyone trying to track down a problem.

Reply:
Fixed